src/include/txn.i

```diff
@@ -999,12 +999,12 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
           cbt->upd_value, false, &cbt->upd_value->buf));
 
     /*
-     * Retry if we race with prepared commit or rollback as the reader may have read changed history
-     * store content.
+     * Retry if we race with prepared rollback as the value the reader should read may have been
+     * removed from the history store to the update chain.
      */
     if (prepare_upd != NULL) {
-        WT_ORDERED_READ(prepare_state, prepare_upd->prepare_state);
-        if (prepare_upd->txnid == WT_TXN_ABORTED || prepare_state == WT_PREPARE_RESOLVED)
```

Comment:
Doesn't this give the problem to a reader who is reading with a timestamp to which he can see the tombstone that gets inserted into the history store as part of the prepare commit?
Scenario:
PU (30) -> U1(20)
PU is written to the data store and U1 is written to the history store. A reader with read timestamp of 50 is trying to read the data at the same time the prepare is committed and a tombstone gets inserted into the history store with timestamp 30. This lead to a key miss.

Reply:
Yes, that's right. I'll revert this change.